### ISSUE TITLE
Questions about reproducing the results

### ISSUE THREAD

**Issue by mprzewie** [zhangq327/U-MAE#11]
Hi! Thank you for your great work! 

I am trying to reproduce your results, but have had no success as of yet. 

For example, when training, and then linear probing on ImageNet-100, I achieve the following results:

| model | top-1 accuracy (linear probing)|
|-|-|
|MAE (pretrained for 200 epochs) | 55.72 |
|U-MAE (pretrained for 200 epochs) |  56.12 |
|MAE (pretrained for 800 epochs)| 73.1 |
|U-MAE (pretrained for 800 epochs)| 75.1|

The above results are from linear probing *after training*. So U-MAE outperforms MAE, but both methods achieve different results from the ones reported in the paper.

This is the command I used for pretraining of U-MAE (for MAE I change `--reg` to `none`)

```
main_pretrain.py \
--batch_size 256 --model mae_vit_base_patch16 --norm_pix_loss \
--mask_ratio 0.75 --epochs 800 --warmup_epochs 40 --blr 1.5e-4 --weight_decay 0.05 \
--num_workers 32 --data_path imagenet100/ \
--lamb 0.01 --reg spectral --norm_pix_loss \
--output_dir out/ --log_dir out/
```

One thing I changed compared to your hyperparameters is the batch size - I'm training on A100 GPUs, so I can fit the batch size of 256 on a GPU. However, I train on two nodes with 8 GPUs each, so in the end the effective batch size should remain the same (4096).

Also, which subset do you use for ImageNet-100? I followed those two works:
* https://proceedings.neurips.cc/paper/2020/hash/4c2e5eaae9152079b9e95845750bb9ab-Abstract.html
* https://arxiv.org/abs/2111.09613
* https://github.com/hankook/AugSelf/blob/main/splits/imagenet100.txt

I would really appreciate your guidance on this.

**Comment by zhangq327** [comment]
Hi!
Thanks for appreciating our work! I think the main difference is that we use different subsets of ImageNet. We select 100 classes following https://www.kaggle.com/datasets/ambityga/imagenet100. Besides, we observe that the results may be slightly different with different GPUs (such as V100 and Rtx 3090). Our original experiments are conducted on V100.

Hope it helps!

**Comment by mprzewie** [comment]
Thanks for the quick reply! Much appreciated! 

I'll try the split you've mentioned. In the meantime, I also have a question about the performance of VIT-B on ImageNet-1K, as I'm also trying to reproduce it and wondering what results to expect.

I saw that there is a discrepancy between the results in your paper (MAE - 55.4% / U-MAE - 58.5%) and the ones posted in the OpenReview discussion (https://openreview.net/forum?id=WOppMAJtvhv&noteId=Phu7f4Cydj), (MAE - 39.72% / U-MAE - 46.49%). From what I understand, in both cases, the models were trained for only 200 epochs. What is the difference between the two sets of results?

**Comment by zhangq327** [comment]
Thanks for your careful reading! In fact, the main difference is that we use different settings of linear probing. For the results (MAE - 39.72% / U-MAE - 46.49%), we train an online linear classifier following (https://github.com/vturrisi/solo-learn) to evaluate the linear probing performance. While in the revision, we find that the online classifier performance is much lower than the offline linear probing in MAE, so we put the online results in Appendix B.5 and update the offline linear probing results in the paper.

Hope it helps!

**Comment by mprzewie** [comment]
Thank you, that makes sense! I found the same thing. 

I'm also wondering whether the batch size is not a possible culprit here.

Based on this script: https://github.com/zhangq327/U-MAE/blob/main/pretrain.sh, and on the paper, I understand that your effective batch size was 1024 (as opposed to 4096 in MAE)? However, in section B.3 you write that you trained the ImageNet model on 4 V100 cards, which seems to be different from the `--nproc_per_node=8` setting in the script and would suggest the effective batch size of 512 (4* 128). So which batch size was used? I'm sorry for such nitpicking, but at this point I think I need to get the hyperparameters exactly right to obtain the correct results.

Once again, I really appreciate your help!

**Comment by zhangq327** [comment]
I'm sorry for the typos. 

Indeed, the effective batch size in the original experiments should be 512. 

Thanks for pointing it out!
